### PR TITLE
Replace uses of the Unicode trait with Str

### PR DESCRIPTION
**Pull request by mdickinson**
This PR replaces all uses of the deprecated `Unicode` trait type with `Str`.

### CODE REVIEW

**Comment by mdickinson** [comment]
I was jumping the gun a bit here. We can't do this until after we've dropped Python 2 support.

**Comment by rahulporuri** [comment]
This PR is ready to go in now because the upcoming release will not support Python 2.

**Comment by kitchoi** [comment]
> This PR is ready to go in now because the upcoming release will not support Python 2.

I am not sure what sequence of events we are going after here for https://github.com/enthought/apptools/issues/120.
e.g. I'd think both `setup.py` and `etstool.py` need to be updated to prevent Python 2 from being used before changes like this PR goes ahead.

**Comment by rahulporuri** [comment]
I think we can merge this after #190 is merged (after fixing merge conflicts of course).

**Review by kitchoi — APPROVED** [review]
LGTM

**Comment by kitchoi** [comment]
I will be resolving the conflicts...

File: apptools/preferences/tests/test_preferences_helper.py
Comment (mdickinson) [inline apptools/preferences/tests/test_preferences_helper.py:195]: We could go further and drop the `u` prefixes, but that's probably best done in a separate PR.

**Review by mdickinson — COMMENTED** [review]
@kitchoi Thank you for the changes! I can't officially approve because I opened the PR in the first place, but LGTM.

**Review by aaronayres35 — APPROVED** [review]
LGTM

**Comment by kitchoi** [comment]
Thank you all!

**Pull request merged**
